pkg/controller/jobs/pod/indexer.go

```diff
@@ -0,0 +1,28 @@
+package pod
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	PodGroupNameCacheKey = "PodGroupNameCacheKey"
+)
+
+func IndexPodGroupName(o client.Object) []string {
+	pod, ok := o.(*corev1.Pod)
+	if !ok {
+		return nil
+	}
+
+	if labelValue, labelExists := pod.GetLabels()[GroupNameLabel]; labelExists {
```

Comment:
nit:
```suggestion
	if labelValue, exists := pod.GetLabels()[GroupNameLabel]; exists {
```

variable name length should be proportional to the scope of the variable. This variable's scope is one line :)